pkg/v22/key/types.go

```diff
@@ -5,8 +5,13 @@ type AppSpec struct {
 	App     string
 	Catalog string
 	Chart   string
-	// Whether app is installed for clusterapi clusters only
-	ClusterAPI      bool
+	// Whether app is installed for clusterapi clusters only.
+	ClusterAPIOnly bool
+	// ConfigMapName overrides the name, otherwise the cluster values configmap
+	// is used.
+	ConfigMapName string
+	// Whether app is installed for legacy clusters only.
+	LegacyOnly      bool
```

Comment:
More prep for Ingress Controller. This is because it is optional for the clusterapi controller.